castle/cms/subscribe.py

```diff
@@ -25,17 +25,29 @@ def __init__(self):
             self._data = OOBTree()
             self.site._subscribers = self._data
 
-    def add(self, email, data=None):
+    def add(self, email, data=None, code=None, categories=None):
+
         email = email.lower()
         if data is None:
             data = {}
-        data.update({
-            'created': time(),
-            'code': make_random_key(100),
-            'confirmed': False,
-            'phone_number_confirmed': False,
-            'email': email
-        })
+        if code is None:
```

Comment:
here we can just do if code is None code = make_random_key and then remove the else clause and just do the data.update() the same either way.  It won't be more efficient cpu-wise to do it that way or anything but, little easier to read / less code.